src/typescript/frontend/src/app/dexscreener/events.ts

```diff
@@ -0,0 +1,308 @@
+  txnId: string;
+  txnIndex: number;
+  eventIndex: number;
+  maker: string;
+  pairId: string;
+  asset0In?: number | string;
+  asset1In?: number | string;
+  asset0Out?: number | string;
+  asset1Out?: number | string;
+  priceNative: number | string;
+  reserves?: {
+    asset0: number | string;
+    asset1: number | string;
+  };
+  metadata?: Record<string, string>;
+}
+
+/**
+ * - `txnId` is a transaction identifier such as a transaction hash
+ * - `txnIndex` refers to the order of a transaction within a block, the higher the index the later in the block the
+ * transaction was processed
+ * - `eventIndex` refer to the order of the event within a transaction, the higher the index the later in the
+ * transaction the event was processed
+ * - The combination of `txnIndex` + `eventIndex` must be unique to any given event within a block, including for
+ * different event types
+ * - `maker` is an identifier for the account responsible for submitting the transaction. In most cases the transaction
+ * is submitted by the same account that sent/received tokens, but in cases where this doesn't apply (i.e.: transaction
+ * submitted by an aggregator) an effort should be made to identify the underlying account.
+ * - All amounts (`assetIn`/`assetOut`/`reserve`) should be decimalized (`amount / (10 ** assetDecimals)`)
+ * - Reserves refer to the pooled amount of each asset *after* a join/exit event has occured. If there are multiple
+ * join/exit events on the same block and reserves cannot be determined after each individual event then it's
+ * acceptable for only the last event to contain the `reserves` prop.
+ * - `metadata` includes any optional auxiliary info not covered in the default schema and not required in most cases
+ */
+export interface JoinExitEvent {
+  eventType: "join" | "exit";
+  txnId: string;
+  txnIndex: number;
+  eventIndex: number;
+  maker: string;
+  pairId: string;
+  amount0: number | string;
+  amount1: number | string;
+  reserves?: {
+    asset0: number | string;
+    asset1: number | string;
+  };
+  metadata?: Record<string, string>;
+}
+
+export type BlockInfo = { block: Block };
+export type Event = (SwapEvent | JoinExitEvent) & BlockInfo;
+
+export interface EventsResponse {
+  events: Event[];
+}
+
+const TWO_TO_64 = new Big(2).pow(64);
+
+export function toDexscreenerSwapEvent(
+  event: ReturnType<typeof toSwapEventModel>
+): SwapEvent & BlockInfo {
+  let assetInOut;
+
+  if (event.swap.isSell) {
+    // We are selling to APT
+    assetInOut = {
+      asset0In: toCoinDecimalString(event.swap.inputAmount, DECIMALS),
+      asset0Out: 0,
+      asset1In: 0,
+      asset1Out: toCoinDecimalString(event.swap.baseVolume, DECIMALS),
+    };
+  } else {
+    // We are buying with APT
+    assetInOut = {
+      asset0In: 0,
+      asset0Out: toCoinDecimalString(event.swap.quoteVolume, DECIMALS),
+      asset1In: toCoinDecimalString(event.swap.inputAmount, DECIMALS),
+      asset1Out: 0,
+    };
+  }
```

Comment:
```suggestion
    assetInOut = {
      asset0In: toCoinDecimalString(event.swap.baseVolume, DECIMALS),
      asset0Out: 0,
      asset1In: 0,
      asset1Out: toCoinDecimalString(event.swap.quoteVolume, DECIMALS),
    };
  } else {
    // We are buying with APT
    assetInOut = {
      asset0In: 0,
      asset0Out: toCoinDecimalString(event.swap.baseVolume, DECIMALS),
      asset1In: toCoinDecimalString(event.swap.quoteVolume, DECIMALS),
      asset1Out: 0,
    };
  }
```
What you'd want to do is, in case of a `sell` and a `buy`:

`sell` means they're exchanging their emojicoins for APT (`base` for `quote`)
`buy` means they're exchanging APT for emojicoins (`quote` for `base`)

so if we treat `asset0/asset1` as `base/quote` respectively:

`sell`, `base` in `quote` out
 - `asset0in` (`base` in) would be `baseVolume`
 - `asset1out` (`quote` out) would be `quoteVolume`
`buy`, `quote` in, `base` out
 - `asset0out` (`base` out) would be `baseVolume`
 - `asset1In` (`quote` in) would be `quoteVolume`

which I think is quite different from the picture, now that I write it out hahaha. I believe I have it correct (in terms of in/out ) but regardless, I'm positive the sell = base for quote and buy = quote for base logic is 100% true